sql/core/src/main/scala/org/apache/spark/sql/jdbc/H2Dialect.scala

```diff
@@ -42,7 +42,7 @@ private[sql] object H2Dialect extends JdbcDialect {
 
   private val distinctUnsupportedAggregateFunctions =
     Set("COVAR_POP", "COVAR_SAMP", "CORR", "REGR_INTERCEPT", "REGR_R2", "REGR_SLOPE", "REGR_SXY",
-      "MODE")
+      "MODE", "PERCENTILE_CONT", "PERCENTILE_DISC")
```

Comment:
why do we need this change? H2 dialect deals with these two functions in `visitInverseDistributionFunction`

Reply:
Because H2 dialect overrides the `visitInverseDistributionFunction` and check with `isSupportedFunction`.

```
     override def visitInverseDistributionFunction(
         funcName: String,
         isDistinct: Boolean,
         inputs: Array[String],
         orderingWithinGroups: Array[String]): String = {
       if (isSupportedFunction(funcName)) {
         super.visitInverseDistributionFunction(
           dialectFunctionName(funcName), isDistinct, inputs, orderingWithinGroups)
       } else {
         throw new UnsupportedOperationException(
           s"${this.getClass.getSimpleName} does not support " +
             s"inverse distribution function: $funcName")
       }
     }
```